addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -220,42 +220,91 @@ export const editorCommands = {
         // element if it's a block then we insert the content in the right places.
         let currentNode = startNode;
         let lastChildNode = false;
+        let firstChildNode = false;
+        const tagULOL = ['OL', 'UL'];
+        let convertList = (node) => {
```

Comment:
This looks to be a function to convert one list type to another. Theoretically, we already have a function that does that which is called `toggleList`. Did you try calling `toggleList` here before recreating a whole new function ? If yes, what didn't work ? If some cases weren't correctly supported, would it be possible to fix `toggleList` to handle the missing cases rather than create a whole new function ?

Reply:
@dmo-odoo  Yes i have tried that using toggleList, and in that code `getTraversedNodes(editor.editable)` always works on the selection / selected item so when we select all the list and toggle it , we get list of element from `getTraversedNodes(editor.editable)` to convert but my case was entirely of pasting it to a new line and in that case the `getTraversedNodes(editor.editable)` will have nothing to convert.Hence i made a separate method for it.

Reply:
Have you tried saving the current selection, putting the selection where you want to call `toggleList`, call it, then restore the saved selection ?

Reply:
Saving the selection but of what?, the paste behavior happens in `insert` method which gets the selection from document of where we want to paste the content not of where we selected and copied.

Reply:
@dmo-odoo  There is one way i can try, i will update the code and will inform you.

Reply:
I think @dmo-odoo meant something like:
```pseudocode
restore = preserveCursor()
setSelection(li)
toggleList()
restore()
```
Did you finally inform him after updating the code?

Reply:
Well maybe not `preserveCursor` as that is for something else but yes, registering the current selection, moving it somewhere else to execute `toggleList` then restore the previous selection. There are methods in history that can help with that like `_computeHistorySelection` and `historyResetLatestComputedSelection`. That being said, it seems the current code does not look like this anymore.